File: ProjetFilRouge/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ProjetFilRouge.Models
{
    public class Category
    {
        public int? IdCategory { get; set; }
        public string NameCategory { get; set; } 

        public Category(int? idCategory,string nameCategory)
        {
            this.IdCategory = idCategory;
            this.NameCategory = nameCategory;
        }

        public Category()
        {
        }
    }
}

File: ProjetFilRougeFront/src/app/_models/category.ts
export class Category{
    private _idCategory: number;
    private _nameCategory: string;

    constructor(name: string = '', id: number = 0){
        this._idCategory = id;
        this._nameCategory = name;
    }

    get idCategory(): number{
        return this._idCategory;
    }

    get nameCategory(): string{
        return this._nameCategory;
    }

    set nameCategory(name: string){
        this._nameCategory = name;
    }
}